File: test_connect.go
package main

import (
	"context"
	"fmt"
	"os"
	"time"

	"github.com/jackc/pgx/v5"
)

func main() {
	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
	defer cancel()

	conn, err := pgx.Connect(ctx, "host=127.0.0.1 port=5432 dbname=test user=test password=test")
	if err != nil {
		fmt.Println("could not connect:", err)
		os.Exit(1)
	}
	defer conn.Close(ctx)

	exec := func(sql string) error {
		_, err := conn.Exec(ctx, sql)
		return err
	}

	fmt.Println("== DROP TABLE (if present) ==")
	if err := exec("DROP TABLE users"); err != nil {
		fmt.Println("none to drop")
	} else {
		fmt.Println("dropped existing")
	}

	fmt.Println("== CREATE TABLE (if not present) ==")
	exec("CREATE TABLE IF NOT PRESENT users (name TEXT, age INT)")
	fmt.Println("ok")

	fmt.Println("== INSERT ==")
	exec("INSERT INTO users (name, age) VALUES ('Alice', 30)")
	exec("INSERT INTO users (name, age) VALUES ('Bob', 25)")
	exec("INSERT INTO users (name, age) VALUES ('Foo', 111)")
	exec("INSERT INTO users (name, age) VALUES ('Foo', 111)")
	exec("INSERT INTO users (name, age) VALUES ('Foo', 111)")
	exec("INSERT INTO users (name, age) VALUES ('Foo', 111)")
	exec("INSERT INTO users (name, age) VALUES ('Bar', 222)")
	exec("INSERT INTO users (name, age) VALUES ('Bar', 222)")
	exec("INSERT INTO users (name, age) VALUES ('Bar', 222)")
	exec("INSERT INTO users (name, age) VALUES ('Bar', 222)")
	fmt.Println("ok")

	fmt.Println("== SELECT ==")
	rows, _ := conn.Query(ctx, "SELECT * FROM users")
	for rows.Next() {
		values, _ := rows.Values()
		fmt.Println(" ", values)
	}
	rows.Close()

	fmt.Println("== SELECT WHERE ==")
	rows, _ = conn.Query(ctx, "SELECT name FROM users WHERE age > 26")
	for rows.Next() {
		values, _ := rows.Values()
		fmt.Println(" ", values)
	}
	rows.Close()

	fmt.Println("== UPDATE ==")
	tag, err := conn.Exec(ctx, "UPDATE users SET age = 31 WHERE name = 'Alice'")
	if err != nil {
		fmt.Println("rows updated:", err)
	} else {
		fmt.Println("rows updated:", tag.RowsAffected())
	}

	fmt.Println("== SELECT AGAIN ==")
	rows, _ = conn.Query(ctx, "SELECT id, name, age FROM users")
	for rows.Next() {
		values, _ := rows.Values()
		fmt.Println(" ", values)
	}
	rows.Close()

	fmt.Println("== DELETE ==")
	tag, err = conn.Exec(ctx, "DELETE FROM users WHERE name = 'Bob'")
	if err != nil {
		fmt.Println("rows deleted:", err)
	} else {
		fmt.Println("rows deleted:", tag.RowsAffected())
	}

	fmt.Println("== FINAL ==")
	rows, _ = conn.Query(ctx, "SELECT * FROM users")
	for rows.Next() {
		values, _ := rows.Values()
		fmt.Println(" ", values)
	}
	rows.Close()

	conn.Close(ctx)
	fmt.Println("DONE")
}

File: test_connect.js
const { Client } = require('pg');

const client = new Client({
    host: '127.0.0.1',
    port: 5432,
    database: 'test',
    user: 'test',
    password: 'test',
    connectionTimeoutMillis: 3000,
});

(async () => {
    try {
        await client.connect();
    } catch (e) {
        console.log(`could not connect: ${e.message}`);
        process.exit(1);
    }

    const exec = async (sql) => {
        const res = await client.query(sql);
        return res;
    };

    console.log('== DROP TABLE (if present) ==');
    try {
        await exec('DROP TABLE users');
        console.log('dropped existing');
    } catch (e) {
        console.log('none to drop');
    }

    console.log('== CREATE TABLE (if not present) ==');
    await exec('CREATE TABLE IF NOT PRESENT users (name TEXT, age INT)');
    console.log('ok');

    console.log('== INSERT ==');
    await exec("INSERT INTO users (name, age) VALUES ('Alice', 30)");
    await exec("INSERT INTO users (name, age) VALUES ('Bob', 25)");
    await exec("INSERT INTO users (name, age) VALUES ('Foo', 111)");
    await exec("INSERT INTO users (name, age) VALUES ('Foo', 111)");
    await exec("INSERT INTO users (name, age) VALUES ('Foo', 111)");
    await exec("INSERT INTO users (name, age) VALUES ('Foo', 111)");
    await exec("INSERT INTO users (name, age) VALUES ('Bar', 222)");
    await exec("INSERT INTO users (name, age) VALUES ('Bar', 222)");
    await exec("INSERT INTO users (name, age) VALUES ('Bar', 222)");
    await exec("INSERT INTO users (name, age) VALUES ('Bar', 222)");
    console.log('ok');

    console.log('== SELECT ==');
    let res = await exec('SELECT * FROM users');
    for (const row of res.rows) {
        console.log(' ', row);
    }

    console.log('== SELECT WHERE ==');
    res = await exec("SELECT name FROM users WHERE age > 26");
    for (const row of res.rows) {
        console.log(' ', row);
    }

    console.log('== UPDATE ==');
    res = await exec("UPDATE users SET age = 31 WHERE name = 'Alice'");
    console.log('rows updated:', res.rowCount);

    console.log('== SELECT AGAIN ==');
    res = await exec('SELECT id, name, age FROM users');
    for (const row of res.rows) {
        console.log(' ', row);
    }

    console.log('== DELETE ==');
    res = await exec("DELETE FROM users WHERE name = 'Bob'");
    console.log('rows deleted:', res.rowCount);

    console.log('== FINAL ==');
    res = await exec('SELECT * FROM users');
    for (const row of res.rows) {
        console.log(' ', row);
    }

    await client.end();
    console.log('DONE');
})();
